modules/mobile/spec/request/threads_request_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../support/helpers/iam_session_helper'
+require_relative '../support/helpers/mobile_sm_client_helper'
+
+RSpec.describe 'Mobile Messages Integration', type: :request do
+  include Mobile::MessagingClientHelper
+  include SchemaMatchers
+
+  let(:user_id) { '10616687' }
+  let(:inbox_id) { 0 }
+  let(:message_id) { 573_059 }
+  let(:va_patient) { true }
+
+  before do
+    allow_any_instance_of(MHVAccountTypeService).to receive(:mhv_account_type).and_return(mhv_account_type)
+    allow(Mobile::V0::Messaging::Client).to receive(:new).and_return(authenticated_client)
+    iam_sign_in(build(:iam_user, iam_mhv_id: '123'))
+  end
+
+  context 'Basic User' do
+    let(:mhv_account_type) { 'Basic' }
+
+    it 'is not authorized' do
+      get "/mobile/v0/messaging/health/folders/#{inbox_id}/threads",
+          headers: iam_headers,
+          params: { page_size: '5', page: '1', sort_field: 'SENDER_NAME', sort_order: 'ASC' }
+      expect(response).not_to be_successful
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  context 'Advanced User' do
```

Comment:
I believe all mobile users are premium users (I forget where I heard this so we may want to verify that ) so I don't think basic and advanced user specs will apply.

Reply:
Nevermind. I just looked up the slack conversation that talked about this and it's not true. we can have MHV advanced and basic users.